resources/qml/Sidebar.qml

```diff
@@ -256,6 +256,44 @@ Rectangle
         }
     }
 
+    Button
+    {
+        id: toggleFilterButton
+
+        anchors.right: parent.right
+        anchors.rightMargin: UM.Theme.getSize("default_margin").width
+        anchors.top: headerSeparator.bottom
+        anchors.topMargin: UM.Theme.getSize("default_margin").height
+
+        height: settingsModeSelection.height
+        width: visible ? height : 0
```

Comment:
Are these to show an animation or something? If it's not visible this shouldn't have any effect, I'd think.